x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -357,18 +368,122 @@ private void createNewProfile(Subject subject, ActionListener<Profile> listener)
                 bulkRequest,
                 TransportSingleItemBulkWriteAction.<IndexResponse>wrapBulkResponse(ActionListener.wrap(indexResponse -> {
                     assert docId.equals(indexResponse.getId());
-                    // TODO: replace with actual domain information
                     final VersionedDocument versionedDocument = new VersionedDocument(
                         profileDocument,
                         indexResponse.getPrimaryTerm(),
                         indexResponse.getSeqNo()
                     );
                     listener.onResponse(versionedDocument.toProfile(Set.of()));
-                }, listener::onFailure))
+                }, e -> {
+                    if (e instanceof VersionConflictEngineException) {
+                        // Document already exists with the specified ID, get the document with the ID
+                        // and check whether it is the right profile for the subject
+                        getOrCreateProfileWithBackoff(subject, profileDocument, DEFAULT_BACKOFF.iterator(), listener);
```

Comment:
@albertzaharovits I decided to keep this to be a GET request instead of Search after some back and forth. Changing it to be a search adds complexity without real gain. Please let me explain:
* We cannot simply re-use the search that locates the existing profile. Because for an empty search result, we cannot tell whether the document is not yet available for search or does not match the search criteria. 
* Alternatively, we can augument the search with extra profile IDs, i.e. let it search for "username + realm/domain" as well "profile IDs" so that we can always get back all potentially relevant profiles. But with this approach, we cannot avoid checking whether "the document is the right one" locally which means we still need separate logic for search and local filtering.

Search is also slower and more expensive than GET. But racing on creating new profile should not happen very often, so it is probably not a big concern.

The only potential advantage that I can think of for using a search is that it has the potential to locate a profile that is being created from a different node with inconsistent configuration. For example, `node-1` has `realm1` without a domain, `node-2` has `realm1` and `realm2` with a domain. If they race to create a new profile, using search can potentially create/locate the same profile if there is an existing profile with clashing UID. The scenario is rather very unlikely. The behaviour seems to be smart. But I am not sure whether we should actively opt-in for it given (1) it's very rare and (2) we probably don't wanna cater for incosistent configuration at this point.

What do you think? I do have another version locally that uses search. So please let me know if you still prefer Search over GET.

Reply:
Using search can also eliminate a few rounds of GET if the differentiator value is big (though with complexities in some other places). It can also potentially support a higher upper limit for the differentiator value (It is now 10 for the GET approach). So another "advantage" to consider.

Reply:
Sounds good to me, thanks for considering it!